pkg/controller/infrastructure/actuator.go

```diff
@@ -507,6 +544,10 @@ func (a *actuator) reconcile(ctx context.Context, infra *extensionsv1alpha1.Infr
 		return err
 	}
 
+	if err = a.ensureOldSSHKeyDetached(ctx, infra); err != nil {
```

Comment:
How will this work - you would always detach the SSH key from the Alicloud machines, right? The key will then only be managed by the `cloud-config-downloader` running on the nodes? If yes, is it possible to not create the machines with the key in the first place, i.e., only use the `cloud-config-downloader` for managing the SSH keys?

Reply:
Good suggestion, I will remove ssh-key generation from terraform and do some other follow up changes and test.